GoogleChromeLabs/ps-analysis-tool
# Feature: Use Chrome DevTools Protocol to extract extra information

## Description
This PR aims to extract information regarding cookies like `partitionKey`, `size`, `priority`, `blocked cookie` and blocked cookie reasoning.
- If a cookie has been blocked then the cookie data row will be highlighted with a background color yellow similar to the network tab. Upon clicking the blocked cookie in the cookie details section there will be a complete description as to why the cookie was blocked.
- Since Chrome DevTools Protocol is being used and activation of `chrome.debugger` shows a notification on the browser a toggle switch has been added to the popup which will allow the user to enable or disable CDP.
- Description of blocking reasons for cookie blocking were not readily available, official MD files and the reasons from the chromium repo were used to show the complete description.
CLI and Extension both use CDP so to simplify some functions have been moved to the `common` package so that only a single copy of those functions exists.
- On the landing page, some brief information regarding the cookies is shown in a similar way as cookie classification was shown. This shows the total blocked cookies, blocked reasons and the count of blocked cookies because of that reason.
- In frame overlay, we show the blocked cookies for the frame and the cumulative cookie blocked reasons for that frame

## Relevant Technical Choices
- Chrome debugger will be attached in the data layer and extra information will be listened to from network requests.
- Existing Chrome API and Chrome debugger will be used together to extract necessary information.

## Testing Instructions
- In the terminal run `npm start`.
- In a fresh instance go to settings and block 3rd party cookies.
- Now open a new tab and go to `aljazerra.com` Notice that the 3rd party cookies have been blocked and now are remarked with a cross under cookie blocked.

## Additional Information:



## Screenshot/Screencast


https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/59614577/69f19df5-88be-4c67-a864-b327ca3f5624





---
Fixes #247
